user-service/controller/user-controller.js

```diff
@@ -1,143 +1,66 @@
-}
-
-export async function updateAccount(req, res) {
-    const { token } = req;
-    const { newUsername, newPassword } = req.body;
-    const { username } = _decodeJwt(token);
+  } catch (_) {
+    return res.status(500).json({ message: 'Server error when creating user.' });
+  }
+};
+
+export const updateUser = async (req, res) => {
+  try {
+    const username = req.params.id; 
+    const { newPassword } = req.body;
     const newProfile = {};
-
     // Only add to the new profile to update to if the value exists
-    if (newUsername) {
-        newProfile.username = newUsername;
-    }
-
     if (newPassword) {
-        newProfile.password = newPassword;
+      newProfile.password = newPassword;
     }
-
+  
     // Directly return if newProfile is empty, no update
     if (
       newProfile && // null and undefined check
       Object.keys(newProfile).length === 0 &&
       Object.getPrototypeOf(newProfile) === Object.prototype
     ) {
-      return res.status(200).json({
-        message: 'No update on your account since no input provided.',
-      });
-    }
-
-    const updatedUser = await _updateAccount(username, newProfile);
-
-    if (updatedUser) {
-      // Create new JWT for updated profile
-      const token = _generateJwt(updatedUser);
-      // Send cookie
-      res.cookie('token', token, { httpOnly: true });
-      return res.status(200).json({ message: 'Updated account successfully.' });
+      return res.status(200).json({ message: 'No update required.' });
     }
-
-    return res.status(409).json({ message: 'Account does not exist / update unsuccessful.' });
-}
-
-export async function deleteAccount(req, res) {
-    const { token } = req;
-    const { username } = _decodeJwt(token);
-    await _blacklistJwt(token);
-    res.clearCookie('token');
-    if (await ormDeleteAccount(username)) {
-        return res.status(200).json({ message: `Deleted account ${username} successfully.` });
-    } else {
-        return res.status(409).json({ message: 'Account does not exist / delete unsuccessful.' });
-    }
-}
-
-export function acknowledgeJWTValidity(req, res) {
-    const { token } = req;
-    const { username } = _decodeJwt(token);
-    return res.status(200).json({ username, message: 'Valid JWT.' });
-}
-
-export async function authenticateJwt(req, res, next) {
-    const token = req.cookies.token;
-    if (!token) {
-        return res.status(401).json({ message: 'No token provided.' });
-    }
-
-    const isBlacklisted = await _getBlacklistedJwt(token);
-    if (isBlacklisted) {
-        return res.status(401).json({ message: 'Please sign in again.' });
-    }
-
-    jwt.verify(token, process.env.JWT_SECRET_KEY, (error, user) => {
-        if (error) {
-            return res.status(401).json({ message: 'Invalid token.' });
-
-        }
-
-        req.token = token;
-        next();
-    })
-}
+  
+    // Update user
+    await _updateUser(username, newProfile);

```

Comment:
What if update fails here? How are you sure that the profile has been updated correctly? I think you should make use of the return result of this function to check first before telling the user it is updated successfully.

Reply:
Logically speaking, if the database does not throw an error when updating the user, I think it is safe to assume that the user is updated correctly, otherwise an error will be thrown and caught by the try-catch loop, which will then display the corresponding error message.